scripts/split-by-solvency.ts

```diff
@@ -0,0 +1,71 @@
+import { SuperTokenList, TokenInfo } from "..";
+import tokenList from "../superfluid.tokenlist.json";
+import fs from "fs";
+import { validate, validateUnderlyingTokens } from "../utils";
+
+const createFilteredList = async (tokenList: SuperTokenList, tier: string) => {
+  const underlyingTokens = tokenList.tokens.filter((token) =>
+    token.tags?.includes("underlying")
+  );
+
+  const filteredByTier = tokenList.tokens.filter((token) =>
+    token.tags?.includes(tier)
+  );
+
+  const tokens: TokenInfo[] = [];
+
+  filteredByTier.forEach((token) => {
+    tokens.push(token);
+
+    if (
+      ["Wrapper", "Native Asset"].includes(
+        token.extensions?.superTokenInfo?.type ?? ""
+      )
+    ) {
+      const underlyingToken = underlyingTokens.find(
+        (underlyingToken) =>
+          underlyingToken.address ===
+          // @ts-ignore (Special Native Assets have underlying tokens)
+          token.extensions?.superTokenInfo?.underlyingTokenAddress
+      );
+      if (underlyingToken) {
+        tokens.push(underlyingToken);
+      }
+    }
+  });
+
+  try {
+    const partialTokenList = { ...tokenList, tokens };
+    await validate(partialTokenList);
+    const result = validateUnderlyingTokens(partialTokenList);
+
+    if (!result) {
+      throw new Error("Underlying tokens validation failed");
+    }
+  } catch (e) {
+    console.error(tier, e);
+  }
+
+  fs.writeFileSync(
+    `solvency-${tier}.tokenlist.json`,
+    JSON.stringify(
+      {
```

Comment:
It's a bit weird code structure to put together a token list, validate it, and then put it together again. I'd prefer the code to flow in a way it's put together once.

Reply:
just an overlook, i was in the middle of refactoring